srcCpp03/RTIDDSImpl.cxx

```diff
@@ -548,7 +600,7 @@ class RTIPublisher: public RTIPublisherBase<T> {
                 }
             }
         } else {
-            key = MAX_CFT_VALUE;
+            key = this->_instance_handles.size()-1;
```

Comment:
Can we double check that this is working properly, not sure why we used MAX_CFT_VALUE, and its value.

Reply:
I am pretty sure this is working as expected. The problem is that I do not know how was that even working before since we were accessing an unallocated memory segment.

Before this change we were doing:
```C++
// In RTIPublisher constructor
for (unsigned long i = 0; i < this->_num_instances; ++i) { // this->_num_instances = 1 by default
    // Register the key of Instance
    this->_instance_handles.push_back(this->_writer.register_instance(this->data));
}

// Register the key of MAX_CFT_VALUE
this->_instance_handles.push_back(this->_writer.register_instance(this->data));

// ...

// In send function
// ...
key = MAX_CFT_VALUE; // MAX_CFT_VALUE = 65535 = [255,255,0,0,]b
//...
if (!isCftWildCardKey) {
    this->_writer.write(this->data, this->_instance_handles[key]);
}
```

So we would be doing `this->_instance_handles[key]` that is `this->_instance_handles[MAX_CFT_VALUE]` which is not accessible since _instance_handles has only two elements, not 65535.

This was causing me some errors during the execution of perftest. With these changes, it works.